src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,443 @@
+  
+  if org == "EPSG"
+    return CoordRefSystems.get(EPSG{coord_id})
+  elseif org == "ESRI"
+    return CoordRefSystems.get(ESRI{coord_id})
+  elseif org == "NONE" || org == ""
+    # NONE organization indicates undefined/custom CRS
+    return Cartesian{NoDatum}
+  else
+    # Other organizations (custom authorities) - use Cartesian as fallback
+    # This branch is reachable for custom organization names
+    return Cartesian{NoDatum}
+  end
+end
+
+function gpkgreadtable(table_result, geom_column::String)
+  # Convert to row iterator that can be collected
+  all_rows = Tables.rowtable(table_result)
+  
+  if isempty(all_rows)
+    return nothing
+  end
+  
+  # Determine attribute names excluding geometry column
+  geom_sym = Symbol(geom_column)
+  attr_names = filter(name -> name != geom_sym, propertynames(first(all_rows)))
+  
+  if isempty(attr_names)
+    return nothing
+  end
+  
+  # Extract attributes from all rows with valid geometry
+  rows = map(all_rows) do row
+    blob = getproperty(row, geom_sym)
+    if !ismissing(blob) && !isnothing(blob)
+      [getproperty(row, name) for name in attr_names]
+    else
+      nothing
+    end
+  end
+  
+  # Filter out nothing values
+  valid_rows = filter(!isnothing, rows)
+  
+  if isempty(valid_rows)
+    return nothing
+  end
+  
+  # Create a named tuple of vectors (Tables.jl compatible)
+  columns = NamedTuple{Tuple(attr_names)}([
+    [row[i] for row in valid_rows] for i in 1:length(attr_names)
+  ])
+  
+  return columns
+end
+
+function gpkgreadgeoms(data_result, geom_column::String, crs_type)
+  geometries = Geometry[]
+  geom_sym = Symbol(geom_column)
+  
+  for row in data_result
+    blob = getproperty(row, geom_sym)
+    if !ismissing(blob) && !isnothing(blob)
+      geom, _ = parse_gpb(blob, crs_type)
+      if !isnothing(geom)
+        push!(geometries, geom)
+      end
+    end
+  end
+  
+  return geometries
+end
+
+function gpkgread(fname::String; layer::Int=1, kwargs...)
+  db = SQLite.DB(fname)
+  
+  # 1. Read metadata to extract names of tables and CRS type
+  # Get feature table names
+  contents_result = DBInterface.execute(db,
+    """
+    SELECT table_name, data_type
+    FROM gpkg_contents
+    WHERE data_type = 'features'
+    ORDER BY table_name
+    """)
+  contents = [row.table_name for row in contents_result]
+  
+  if isempty(contents)
+    close(db)
+    error("No vector feature tables found in GeoPackage")
+  end
+  
+  if layer > length(contents)
+    close(db)
+    error("Layer $layer not found. GeoPackage contains $(length(contents)) feature layers")
+  end
+  
+  table_name = contents[layer]
+  
+  # Get geometry column information
+  geom_result = DBInterface.execute(db,
+    """
+    SELECT column_name, srs_id
+    FROM gpkg_geometry_columns
+    WHERE table_name = ?
+    """, [table_name])
+  geom_info = first(geom_result)
+  geom_column = geom_info.column_name
+  srs_id = Int32(geom_info.srs_id)
+  crs_type = get_crs_from_srid(db, srs_id)
```

Comment:
I believe this entire section deserves a separate function that take the `db` and `layer` as inputs and outputs the relevant information: `table_name`, `geom_column` and `crs_type`.